src/test/java/teammates/test/cases/ui/pagedata/StudentCourseDetailsPageDataTest.java

```diff
@@ -0,0 +1,168 @@
+        assertFalse(getInstructors(pageData).isEmpty());        
+        assertFalse(getTeammates(pageData).isEmpty());
+        
+        assertEquals(student.email, pageData.getStudentCourseDetailsPanel().getStudentEmail());
+        assertEquals(student.name, pageData.getStudentCourseDetailsPanel().getStudentName());
+        assertEquals(team.name, pageData.getStudentCourseDetailsPanel().getStudentTeam());
+        
+        assertEquals(1, getInstructors(pageData).size());
+        assertEquals("Instructor 5 of CourseNoRegister", getInstructors(pageData).get(0).name);
+        assertEquals("Instructor", getInstructors(pageData).get(0).displayedName);
+        
+        assertEquals(1, getTeammates(pageData).size());
+        
+        ______TS("student in archived course");
+              
+        student = dataBundle.students.get("student1InArchivedCourse");        
+        course = dataBundle.courses.get("archivedCourse");
+        
+        courseDetails = new CourseDetailsBundle(course);
+        
+        instructors = new ArrayList<InstructorAttributes>();
+        instructors.add(dataBundle.instructors.get("instructorOfArchivedCourse"));
+        
+        team = new TeamDetailsBundle();
+        team.name = student.team;
+        
+        // Get team members of student
+        for (Entry<String, StudentAttributes> entry : dataBundle.students.entrySet()) {
+            StudentAttributes currStudent = entry.getValue();
+            if (currStudent.team.equals(team.name)) {
+                team.students.add(currStudent);
+            }
+        }
+        
+        pageData.init(courseDetails, instructors, student, team);
+        
+        assertEquals(courseDetails.course.name, pageData.getStudentCourseDetailsPanel().getCourseName());
+        assertEquals(courseDetails.course.id, pageData.getStudentCourseDetailsPanel().getCourseId());
+        
+        assertFalse(getInstructors(pageData).isEmpty());        
+        assertFalse(getTeammates(pageData).isEmpty());
+        
+        assertEquals(student.email, pageData.getStudentCourseDetailsPanel().getStudentEmail());
+        assertEquals(student.name, pageData.getStudentCourseDetailsPanel().getStudentName());
+        assertEquals(team.name, pageData.getStudentCourseDetailsPanel().getStudentTeam());
+        
+        assertEquals(1, getInstructors(pageData).size());
+        assertEquals("InstructorOfArchiveCourse name", getInstructors(pageData).get(0).name);
+        assertEquals("Instructor", getInstructors(pageData).get(0).displayedName);
+        
+        assertEquals(3, getTeammates(pageData).size());
+    }
+    
+    public List<InstructorAttributes> getInstructors(StudentCourseDetailsPageData pageData) {
+        return pageData.getStudentCourseDetailsPanel().getInstructors();
+    }
+    
+    public List<StudentAttributes> getTeammates(StudentCourseDetailsPageData pageData) {
+        return pageData.getStudentCourseDetailsPanel().getTeammates();
+    }
```

Comment:
Mentioned this in the previous PR too, but is this better than storing it as a variable? Why?